editor/modes/visual-editor/block.js

```diff
@@ -251,6 +298,18 @@ class VisualEditorBlock extends Component {
 	onKeyDown( event ) {
 		const { keyCode, target } = event;
 
+		this.handleToolbarTabCycle( event );
+		this.handleToolbarArrowCycle( event );
+
+		if ( keyCode === F10 && ! ( event.altKey || event.ctrlKey || event.shiftKey || event.metaKey ) ) {
+			event.preventDefault();
+			event.stopPropagation();
+			if ( this.props.isTyping ) {
+				this.props.onStopTyping();
+			}
+			this.props.onFocus( this.props.uid, { toolbar: true } );
```

Comment:
The problem is that the toolbar is hidden most of the time so it must be shown before it can be focused. Since I had to use react to show it it seemed simplest to focus it using the react lifecycle.